ChatRPG/Services/EmailSender.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Drawing.Printing;
+using Microsoft.AspNetCore.Identity.UI.Services;
+using MailKit.Net.Smtp;
+using MimeKit;
+using MimeKit.Text;
+
+namespace ChatRPG.Services;
+
+public class EmailSender : IEmailSender
+{
+
+    private readonly string _senderEmail;
+    private readonly string _senderPassword;
+
+    public EmailSender(IConfiguration configuration)
+    {
+        _senderEmail = configuration.GetSection("ChatRPGEmail").GetValue<string>("Email");
+        _senderPassword = configuration.GetSection("ChatRPGEmail").GetValue<string>("Password");
+    }
+
+    public async Task SendEmailAsync(string email, string subject, string htmlMessage)
+    {
+        MimeMessage mail = new MimeMessage();
+        mail.From.Add(new MailboxAddress("ChatRPG", _senderEmail));
+        mail.To.Add(new MailboxAddress(email, email));
+        mail.Subject = subject;
+        mail.Body = new TextPart(TextFormat.Html)
+        {
+            Text = htmlMessage
+        };
+
+        SmtpClient smtpClient = new SmtpClient();
+        await smtpClient.ConnectAsync("smtp.gmail.com", 0, true);
+        await smtpClient.AuthenticateAsync(_senderEmail, _senderPassword);
+        await smtpClient.SendAsync(mail);
+        await smtpClient.DisconnectAsync(true);
```

Comment:
Skal der håndteres nogle exceptions her?

Reply:
Yeppers, det har du ret i. Den crasher vidst hvis at Emailen ikke er sat i user-secrets. Men hvad der skal gøres kommer an på hvad vi bestemmer ift kommentaren ovenover

Reply:
Nu hvor jeg tænker over det, så tror jeg ikke det er nødvendigt. Altså programmet kan ikke recover hvis der kommer en exception her (feks fordi at email credentials ikke er valid). Især nu hvor jeg har tilføjet tjekket om der skal sendes en mail eller ej. 
Medmindre du har noget andet i tankerne?

Reply:
Yeah - så vil jeg nok foretrække at fange exceptionen og logge den. Dvs.:
```csharp
try
{
    // do mail stuff
}
catch (EmailException e) // eller whatever exception der kan blive kastet
{
    _logger.LogError("Failed to send email", e);
}
```
hvor `_logger` er en `ILogger<EmailSender>` som injectes gennem constructoren